tools/src/commands/GenerateBareApp.ts

```diff
@@ -0,0 +1,133 @@
+
+  const repoRoot = path.resolve(__dirname, '../../../');
+  const projectsDir = path.resolve(process.cwd(), outDir);
+  const projectDir = path.resolve(process.cwd(), projectsDir, appName);
+
+  if (!fs.existsSync(projectsDir)) {
+    fs.mkdirSync(projectsDir);
+  }
+
+  if (clean) {
+    if (fs.existsSync(projectDir)) {
+      fs.rmdirSync(projectDir, { recursive: true });
+    }
+  }
+
+  // these don't seem to symlink properly w/ expo-yarn-workspaces
+  const symlinkedPackages = ['expo', 'expo-modules-autolinking'];
+
+  // these seem to be included regardless of what is in node_modules (default packages for prebuild)
+  const defaultPackages = [
+    'expo-asset',
+    'expo-application',
+    'expo-constants',
+    'expo-file-system',
+    'expo-font',
+    'expo-keep-awake',
+    'expo-error-recovery',
+    'expo-modules-core',
+    'expo-splash-screen',
+    'expo-updates',
+    'expo-dev-client',
+    'expo-dev-launcher',
+    'expo-dev-menu',
+    'expo-dev-menu-interface',
+  ];
+
+  await runExpoCliAsync('init', [appName, '--no-install', '--template', template], {
+    cwd: projectsDir,
+  });
+
+  const pkg = require(path.resolve(projectDir, 'package.json'));
+
+  pkg['expo'] = pkg['expo'] ?? {};
+
+  // TODO - autolinking -> { include?: [] }
+  // pkg['expo']['autolinking'] = { exclude: [] };
+
+  pkg['scripts']['postinstall'] = 'expo-yarn-workspaces postinstall';
+  pkg['main'] = '__generated__/AppEntry.js';
+
+  pkg['expo-yarn-workspaces'] = {};
+  pkg['expo-yarn-workspaces']['symlinks'] = [];
+
+  packageNames.forEach((packageName) => {
+    pkg.dependencies[packageName] = '*';
+    pkg['expo-yarn-workspaces']['symlinks'].push(packageName);
+  });
+
+  defaultPackages.forEach((packageName) => {
+    pkg.dependencies[packageName] = '*';
+    pkg['expo-yarn-workspaces']['symlinks'].push(packageName);
+  });
+
+  fs.writeJsonSync(path.resolve(projectDir, 'package.json'), pkg);
+
+  console.log('Yarning');
+  await spawnAsync('yarn', [], { cwd: projectDir });
+
+  symlinkedPackages.forEach((packageName) => {
+    symlinkPackageManually(packageName);
+  });
+
+  await runExpoCliAsync('prebuild', ['--clean'], { cwd: projectDir });
+
+  const ncl = path.resolve(repoRoot, 'apps/native-component-list');
+
+  fs.copyFileSync(
+    path.resolve(ncl, 'metro.config.js'),
+    path.resolve(projectDir, 'metro.config.js')
+  );
+
+  fs.copyFileSync(
+    path.resolve(ncl, 'metro.transformer.js'),
+    path.resolve(projectDir, 'metro.transformer.js')
+  );
+
+  // for some reason prebuild is updating the gradle.properties FLIPPER_VERSION which causes SoLoader crash on launch
+  const gradlePropertiesPath = path.resolve(projectDir, 'android', 'gradle.properties');
+  const gradleProperties = fs.readFileSync(gradlePropertiesPath, { encoding: 'utf-8' });
+  const updatedGradleProperies = gradleProperties.replace(
+    `FLIPPER_VERSION=0.54.0`,
+    `FLIPPER_VERSION=0.99.0`
```

Comment:
Could you add a comment why?

Reply:
do you mean write a comment in the `gradle.properties` file? I added a comment just above this for why I needed to write to this file in case you missed - prebuild is updating it for some reason to an old flipper value

Reply:
Ah, gotcha! Sorry, I missed that comment 🙈